app/views/handlers/list_action.py

```diff
@@ -52,21 +52,31 @@ def get_next_location_url(self):
             ):
                 return self.get_section_summary_url()
 
+        if self.router.is_block_complete(
+            block_id=self.parent_location.block_id,
+            section_id=self.parent_location.section_id,
+            list_item_id=self.parent_location.list_item_id,
+        ):
+            return self.router.get_next_location_url(
+                self.parent_location,
+                self._routing_path,
+                self._return_to,
+                self._return_to_answer_id,
+                self._return_to_block_id,
+            )
+
         return self.parent_location.url()
```

Comment:
This will need to persist the query params. A test will need to be added to ensure this is working.

```py
        return self.parent_location.url(
            return_to=self._return_to,
            return_to_answer_id=self._return_to_answer_id,
            return_to_block_id=self._return_to_block_id,
        )
```

Reply:
Query params added